// Copyright (c) 2010-2021 The Open-Transactions developers
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at http://mozilla.org/MPL/2.0/.

#ifndef OPENTXS_PROTOBUF_AUTHORITY_HPP
#define OPENTXS_PROTOBUF_AUTHORITY_HPP

#include "opentxs/Version.hpp"  // IWYU pragma: associated

#include "opentxs/protobuf/Enums.pb.h"

namespace opentxs
{
namespace proto
{
class Authority;
}  // namespace proto
}  // namespace opentxs

namespace opentxs
{
namespace proto
{
auto CheckProto_1(
    const Authority& input,
    const bool silent,
    const std::string& nymID,
    const KeyMode& key,
    bool& haveHD,
    const AuthorityMode& mode = AUTHORITYMODE_ERROR) -> bool;
auto CheckProto_2(
    const Authority& input,
    const bool silent,
    const std::string& nymID,
    const KeyMode& key,
    bool& haveHD,
    const AuthorityMode& mode = AUTHORITYMODE_ERROR) -> bool;
auto CheckProto_3(
    const Authority& input,
    const bool silent,
    const std::string& nymID,
    const KeyMode& key,
    bool& haveHD,
    const AuthorityMode& mode = AUTHORITYMODE_ERROR) -> bool;
auto CheckProto_4(
    const Authority& input,
    const bool silent,
    const std::string& nymID,
    const KeyMode& key,
    bool& haveHD,
    const AuthorityMode& mode = AUTHORITYMODE_ERROR) -> bool;
auto CheckProto_5(
    const Authority& input,
    const bool silent,
    const std::string& nymID,
    const KeyMode& key,
    bool& haveHD,
    const AuthorityMode& mode = AUTHORITYMODE_ERROR) -> bool;
auto CheckProto_6(
    const Authority& input,
    const bool silent,
    const std::string& nymID,
    const KeyMode& key,
    bool& haveHD,
    const AuthorityMode& mode = AUTHORITYMODE_ERROR) -> bool;
auto CheckProto_7(
    const Authority& input,
    const bool silent,
    const std::string& nymID,
    const KeyMode& key,
    bool& haveHD,
    const AuthorityMode& mode = AUTHORITYMODE_ERROR) -> bool;
auto CheckProto_8(
    const Authority& input,
    const bool silent,
    const std::string& nymID,
    const KeyMode& key,
    bool& haveHD,
    const AuthorityMode& mode = AUTHORITYMODE_ERROR) -> bool;
auto CheckProto_9(
    const Authority& input,
    const bool silent,
    const std::string& nymID,
    const KeyMode& key,
    bool& haveHD,
    const AuthorityMode& mode = AUTHORITYMODE_ERROR) -> bool;
auto CheckProto_10(
    const Authority& input,
    const bool silent,
    const std::string& nymID,
    const KeyMode& key,
    bool& haveHD,
    const AuthorityMode& mode = AUTHORITYMODE_ERROR) -> bool;
auto CheckProto_11(
    const Authority& input,
    const bool silent,
    const std::string& nymID,
    const KeyMode& key,
    bool& haveHD,
    const AuthorityMode& mode = AUTHORITYMODE_ERROR) -> bool;
auto CheckProto_12(
    const Authority& input,
    const bool silent,
    const std::string& nymID,
    const KeyMode& key,
    bool& haveHD,
    const AuthorityMode& mode = AUTHORITYMODE_ERROR) -> bool;
auto CheckProto_13(
    const Authority& input,
    const bool silent,
    const std::string& nymID,
    const KeyMode& key,
    bool& haveHD,
    const AuthorityMode& mode = AUTHORITYMODE_ERROR) -> bool;
auto CheckProto_14(
    const Authority& input,
    const bool silent,
    const std::string& nymID,
    const KeyMode& key,
    bool& haveHD,
    const AuthorityMode& mode = AUTHORITYMODE_ERROR) -> bool;
auto CheckProto_15(
    const Authority& input,
    const bool silent,
    const std::string& nymID,
    const KeyMode& key,
    bool& haveHD,
    const AuthorityMode& mode = AUTHORITYMODE_ERROR) -> bool;
auto CheckProto_16(
    const Authority& input,
    const bool silent,
    const std::string& nymID,
    const KeyMode& key,
    bool& haveHD,
    const AuthorityMode& mode = AUTHORITYMODE_ERROR) -> bool;
auto CheckProto_17(
    const Authority& input,
    const bool silent,
    const std::string& nymID,
    const KeyMode& key,
    bool& haveHD,
    const AuthorityMode& mode = AUTHORITYMODE_ERROR) -> bool;
auto CheckProto_18(
    const Authority& input,
    const bool silent,
    const std::string& nymID,
    const KeyMode& key,
    bool& haveHD,
    const AuthorityMode& mode = AUTHORITYMODE_ERROR) -> bool;
auto CheckProto_19(
    const Authority& input,
    const bool silent,
    const std::string& nymID,
    const KeyMode& key,
    bool& haveHD,
    const AuthorityMode& mode = AUTHORITYMODE_ERROR) -> bool;
auto CheckProto_20(
    const Authority& input,
    const bool silent,
    const std::string& nymID,
    const KeyMode& key,
    bool& haveHD,
    const AuthorityMode& mode = AUTHORITYMODE_ERROR) -> bool;
}  // namespace proto
}  // namespace opentxs
#endif  // OPENTXS_PROTOBUF_AUTHORITY_HPP
